debugd/internal/debugd/logcollector/logcollector.go

```diff
@@ -152,6 +167,51 @@ func getTemplate(ctx context.Context, logger *logger.Logger) (*template.Template
 	return tmpl, nil
 }
 
+func getFilebeatTemplate(ctx context.Context, logger *logger.Logger) (*template.Template, error) {
+	createContainerArgs := []string{
+		"create",
+		"--name=template",
+		versions.FilebeatImage,
+	}
+	createContainerCmd := exec.CommandContext(ctx, "podman", createContainerArgs...)
+	logger.Infof("Creating filebeat template container")
+	if out, err := createContainerCmd.CombinedOutput(); err != nil {
+		return nil, fmt.Errorf("creating filebeat template container: %w\n%s", err, out)
+	}
+
+	if err := os.MkdirAll("/run/filebeat", 0o777); err != nil {
+		return nil, fmt.Errorf("creating filebeat template dir: %w", err)
+	}
+
+	copyFromArgs := []string{
+		"cp",
+		"template:/usr/share/constellogs/templates/",
+		"/run/filebeat/",
+	}
+	copyFromCmd := exec.CommandContext(ctx, "podman", copyFromArgs...)
+	logger.Infof("Copying filebeat templates")
+	if out, err := copyFromCmd.CombinedOutput(); err != nil {
+		return nil, fmt.Errorf("copying filebeat templates: %w\n%s", err, out)
+	}
+
+	removeContainerArgs := []string{
+		"rm",
+		"template",
+	}
+	removeContainerCmd := exec.CommandContext(ctx, "podman", removeContainerArgs...)
+	logger.Infof("Removing filebeat template container")
+	if out, err := removeContainerCmd.CombinedOutput(); err != nil {
+		return nil, fmt.Errorf("removing filebeat template container: %w\n%s", err, out)
+	}
+
+	tmpl, err := template.ParseFiles("/run/filebeat/templates/filebeat.yml")
+	if err != nil {
+		return nil, fmt.Errorf("parsing filebeat template: %w", err)
+	}
+
+	return tmpl, nil
+}
```

Comment:
maybe you could reuse the `getLogstashTemplate` function and pass the paths you want to copy? Seems pretty generic...

Reply:
I don't really see the value here. I think a generic implementation would be way more complicated to read, as we would have a lot of function args, considering that we template the files before copying them. Also, we change individual yaml fields of the files. If you have an idea for an easy implementation im up to do it though